ui/js/services/pinboard.js

```diff
@@ -24,9 +24,9 @@ treeherder.factory('thPinboard', [
 
                 classification.job_id = job.id;
                 classification.create()
-                    .success(function () {
+                    .then(() => {
                         thNotify.send("Classification saved for " + job.platform + " " + job.job_type_name, "success");
-                    }).error(function (data) {
+                    }).catch(({ data }) => {
```

Comment:
In this case ``data`` should not be deconstructed.  When we pass it to the ``ThModelErrors.format`` on line 53, it tries to fetch ``.data`` out of it.  So we should just use ``catch((data)) =>``  or ``response``, if that seems more correct.

Reply:
Ah good spot :-)

I think this is also broken on master from inspection and testing using `yarn start` and the trick above? (This may be what you meant; happy to fix it whilst I'm here either way)